replace_with/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+// Julien Chien <jchien17@cmc.edu>
+
+pub fn replace_with<T, F: FnOnce(T) -> T>(t: &mut T, f: F) {
+  unsafe {
+    let mut ptr: *mut T = t;
+    *ptr = f(std::ptr::read(ptr))
```

Comment:
Oh, weird. When I run `cargo test --test required` it says both complex and simple type tests pass. Am I running the tests incorrectly?